widgetry/src/widgets/stash.rs

```diff
@@ -0,0 +1,34 @@
+use std::cell::RefCell;
+use std::rc::Rc;
+
+use crate::{EventCtx, GfxCtx, ScreenDims, ScreenPt, Widget, WidgetImpl, WidgetOutput};
+
+/// An invisible widget that stores some arbitrary data on the Panel. Users of the panel can read
+/// and write the value. This is one method for "returning" data when a State completes.
+pub struct Stash<T> {
+    value: Rc<RefCell<T>>,
+}
+
+impl<T: 'static> Stash<T> {
+    pub fn new(name: &str, value: T) -> Widget {
+        Widget::new(Box::new(Stash {
+            value: Rc::new(RefCell::new(value)),
+        }))
+        .named(name)
+    }
+
+    pub(crate) fn get_value(&self) -> Rc<RefCell<T>> {
+        self.value.clone()
+    }
+}
+
+impl<T: 'static> WidgetImpl for Stash<T> {
```

Comment:
I understand why it's convenient, but it is surprising that a "Widget" would be permanently invisible and exist only as a way to store state.

I think in any other framework/language, I'd expect to see some kind of callback or delegate relationship, where the `Selector` notifies its creator (`TripTable` in this case) what its new selection is.

But I guess that's difficult because of Rust or Widgetry... I don't have any other solution at hand, and it sounds like you've put some thought into alternatives, so I say go for it.

Reply:
Keeping UI widgets/controls and application state synchronized seems like the crux of all Rust GUI libraries. The stash is definitely strange, but it's at least kind of consistent with how widgetry tends to work -- application code explicitly asks for the current state or for changes. Want to know if a button was clicked? Hand the panel the event, see if the outcome was clicked. Need the current checkbox/slider state? Just query.

Grabbing the current stashed value fits that mold; the kind of weird new thing is handing out a reference-counted way to mutate the value in a different `State` than the one with the owner panel. I don't know how to express the lifetime of that stashed value in Rust, but I can at least verbally explain it -- it belongs to the `State` that owns the panel, and `States` live on a stack.